From d5a08c16606a0a9422abf099eeb2c9846af370b0 Mon Sep 17 00:00:00 2001
From: David Andrey <david.andrey@netmodule.com>
Date: Wed, 28 Nov 2012 17:33:47 +0100
Subject: [PATCH 07/17] ARM: xilinx: EMACPS: quick fix for overrun deadlock

RX patch was frozen if no new frame was detected and the
BDRing was full

Signed-off-by:David Andrey david.andrey@netmodule.com
---
 drivers/net/ethernet/xilinx/xilinx_emacps.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_emacps.c b/drivers/net/ethernet/xilinx/xilinx_emacps.c
index d1faef5..70831c4 100644
--- a/drivers/net/ethernet/xilinx/xilinx_emacps.c
+++ b/drivers/net/ethernet/xilinx/xilinx_emacps.c
@@ -1678,7 +1678,8 @@ static int xemacps_rx_poll(struct napi_struct *napi, int budget)
 		dev_dbg(&lp->pdev->dev, "poll RX status 0x%x weight 0x%x\n",
 			regval, budget);
 
-		if (!(regval & XEMACPS_RXSR_FRAMERX_MASK)) {
+		/* if (!(regval & XEMACPS_RXSR_FRAMERX_MASK)) { */
+		if ((regval == 0)) {	/* @@NetModule, rs/da, check with Xilinx */
 			dev_dbg(&lp->pdev->dev, "No RX complete status 0x%x\n",
 				regval);
 			napi_complete(napi);
-- 
1.7.4.1

